userspace/libsinsp/threadinfo.cpp

```diff
@@ -1171,11 +1185,11 @@ void sinsp_threadinfo::cgroups_to_iovec(struct iovec **iov, int *iovcnt,
 	// We allocate an iovec big enough to hold all the cgroups and
 	// intermediate '=' signs. Based on alen, we might not use all
 	// of the iovec.
-	*iov = (struct iovec *) malloc((3 * m_cgroups.size()) * sizeof(struct iovec));
+	*iov = (struct iovec *) malloc((3 * cgroups().size()) * sizeof(struct iovec));
 
 	*iovcnt = 0;
 
-	for(auto it = m_cgroups.begin(); it != m_cgroups.end() && alen > 0; ++it)
+	for(auto it = cgroups().begin(); it != cgroups().end() && alen > 0; ++it)
```

Comment:
I'd feel better about this if we didn't call `cgroups()` repeatedly here. What if it changes from one call to another?

(same for all the other places where we iterate over cgroups)

Reply:
I had the same feeling while doing it, but I convinced myself the compiler would be smart enough to make a single call.
But I'd say I'll change to rage loop anyway.